extensions/wikia/Pages/UpdatePagesTask.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+use Wikia\Tasks\Tasks\BaseTask;
+
+class UpdatePagesTask extends BaseTask {
+	/**
+	 * Delete the entry for this page from dataware.pages table.
+	 *
+	 * @param int $articleId
+	 * @throws DBUnexpectedError
+	 * @throws MWException
+	 */
+	public function deleteEntry( int $articleId ) {
+		$whereCondition = [
+			'page_wikia_id' => $this->getWikiId(),
+			'page_id' => $articleId
+		];
+
+		$this->getDatawareMaster()
+			->delete( 'pages', $whereCondition, __METHOD__ );
+	}
+
+	/**
+	 * Insert a row for this article into dataware.pages table, or update the row if it already exists.
+	 *
+	 * @param int $articleId
+	 * @param bool $isContentPage
+	 * @throws Exception
+	 */
+	public function insertOrUpdateEntry( int $articleId, bool $isContentPage ) {
+		$dbr = wfGetDB( DB_SLAVE );
+
+		// We will need all information about this page plus the timestamp of the last edit made to it
+		// Fetch that using the ID of the latest revision from page_latest
+		$rowToInsert = (array) $dbr->selectRow(
+			[ 'page', 'revision' ], [
+				'page_id',
+				'page_namespace',
+				'page_title',
+				'page_is_redirect',
+				'page_latest',
+				'rev_timestamp AS page_last_edited',
+			],
+			[ 'page_id' => $articleId ],
+			__METHOD__,
+			[],
+			[ 'revision' => [ 'INNER JOIN', [ 'page_latest = rev_id' ] ] ]
+		);
```

Comment:
We risk a slave lag here. Can we pass this information as a task parameters and get them when the task is queued?